examples/pytorch/ogb/directional_GSN/main.py

```diff
@@ -0,0 +1,433 @@
+from ogb.graphproppred import Evaluator
+import torch
+import numpy as np
+from dgl.dataloading import GraphDataLoader
+from tqdm import tqdm
+import dgl
+import random
+import torch.nn as nn
+from ogb.graphproppred.mol_encoder import AtomEncoder
+import torch.nn.functional as F
+from functools import partial
+import torch.optim as optim
+import argparse
+from torch.utils.data import Dataset
+from dgl.data.utils import Subset
+from preprocessing import prepare_dataset
+
+def aggregate_mean(h, vector_field, h_in):
+    return torch.mean(h, dim=1)
+
+def aggregate_max(h, vector_field, h_in):
+    return torch.max(h, dim=1)[0]
+
+def aggregate_sum(h, vector_field, h_in):
+    return torch.sum(h, dim=1)
+
+def aggregate_dir_dx(h, vector_field, h_in, eig_idx):
+    eig_w = ((vector_field[:, :, eig_idx]) /
+             (torch.sum(torch.abs(vector_field[:, :, eig_idx]), keepdim=True, dim=1) + 1e-8)).unsqueeze(-1)
+    h_mod = torch.mul(h, eig_w)
+    return torch.abs(torch.sum(h_mod, dim=1) - torch.sum(eig_w, dim=1) * h_in)
+
+AGGREGATORS = {'mean': aggregate_mean, 'sum': aggregate_sum, 'max': aggregate_max,
+               'dir1-dx': partial(aggregate_dir_dx, eig_idx=1)}
+
+class MLP(nn.Module):
+    def __init__(self, in_size, hidden_size, out_size, 
```

Comment:
`hidden_size` and `out_size` are always the same. You only need to keep one of them.